src/chunkformatter.js

```diff
@@ -0,0 +1,311 @@
+ */
+ChunkFormatter.prototype.moveToNextState = function(chunk, callback) {
+  const row = this.row;
+  if (chunk.commitRow) {
+    callback(null, row);
+    this.commit();
+  } else {
+    if (chunk.valueSize > 0) {
+      this.state = RowStateEnum.CELL_IN_PROGRESS;
+    } else {
+      this.state = RowStateEnum.ROW_IN_PROGRESS;
+    }
+  }
+};
+/**
+ * Process chunk when in NEW_ROW state.
+ * @private
+ * @param {chunks} chunk chunk to process
+ * @param {options} options options
+ * @param {callback} callback callback to call with row as and when generates
+ */
+ChunkFormatter.prototype.newRow = function(chunk, options, callback) {
+  const newRowKey = Mutation.convertFromBytes(chunk.rowKey);
+  this.validateNewRow(chunk);
+  const row = this.row;
+  row.key = newRowKey;
+  row.data = {};
+  this.family = row.data[chunk.familyName.value] = {};
+  const qualifierName = Mutation.convertFromBytes(chunk.qualifier.value);
+  this.qualifiers = this.family[qualifierName] = [];
+  this.qualifier = {
+    value: Mutation.convertFromBytes(chunk.value, options),
+    labels: chunk.labels,
+    timestamp: chunk.timestampMicros,
+    size: chunk.valueSize,
+  };
+  this.qualifiers.push(this.qualifier);
+  this.moveToNextState(chunk, callback);
+};
+/**
+ * Process chunk when in ROW_IN_PROGRESS state.
+ * @private
+ * @param {chunk} chunk chunk to process
+ * @param {*} options  option
+ * @param {*} callback callback to call with row as and when generates
+ */
+ChunkFormatter.prototype.rowInProgress = function(chunk, options, callback) {
+  this.validateRowInProgress(chunk);
+  if (chunk.resetRow) {
+    return this.reset();
+  }
+  const row = this.row;
+  if (chunk.familyName) {
+    this.family = row.data[chunk.familyName.value] =
+      row.data[chunk.familyName.value] || {};
+  }
+  if (chunk.qualifier) {
+    const qualifierName = Mutation.convertFromBytes(chunk.qualifier.value);
+    this.qualifiers = this.family[qualifierName] =
+      this.family[qualifierName] || [];
+  }
+  this.qualifier = {
+    value: Mutation.convertFromBytes(chunk.value, options),
+    labels: chunk.labels,
+    timestamp: chunk.timestampMicros,
+    size: chunk.valueSize,
+  };
+  this.qualifiers.push(this.qualifier);
+  this.moveToNextState(chunk, callback);
+};
+/**
+ * Process chunk when in CELl_IN_PROGRESS state.
+ * @private
+ * @param {chunk} chunk chunk to process
+ * @param {options} options options
+ * @param {callback} callback callback to call with row as and when generates
+ */
+ChunkFormatter.prototype.cellInProgress = function(chunk, options, callback) {
+  this.validateCellInProgress(chunk);
+  if (chunk.resetRow) {
+    return this.reset();
+  }
+  this.qualifier.value =
+    this.qualifier.value + Mutation.convertFromBytes(chunk.value, options);
+  this.qualifier.size = chunk.valueSize;
+  this.moveToNextState(chunk, callback);
+};
+
+module.exports = ChunkFormatter;
+module.exports.RowStateEnum = RowStateEnum;
```

Comment:
Is there a reason to export RowStateEnum? Shouldn't this be private?

Reply:
I wanted to use it in test case, since its just based on number, I could have created new fake for enum in test case. I think its ok to export this, as this is immutable. But I will wait for more feedback if we shouldnt export this.